conformance/client/src/main/kotlin/com/connectrpc/conformance/client/ClientArgs.kt

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022-2023 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.connectrpc.conformance.client
+
+import com.connectrpc.conformance.client.adapt.UnaryClient.InvokeStyle
+
+data class ClientArgs(
+    val invokeStyle: InvokeStyle,
+    val verbosity: Int,
+) {
+    companion object {
+        fun parseArgs(args: Array<String>): ClientArgs {
+            var invokeStyle = InvokeStyle.SUSPEND
+            var verbosity = 0
+            var skip = false
+            for (i in args.indices) {
+                if (skip) {
+                    skip = false
+                    continue
+                }
+                when (val arg = args[i]) {
+                    "-s", "--style" -> {
+                        if (i == args.size - 1) {
+                            throw RuntimeException("$arg option requires a value")
+                        }
+                        skip = true // consuming next string now
+                        val v = args[i + 1]
+                        when (v.lowercase()) {
+                            "suspend" -> {
+                                invokeStyle = InvokeStyle.SUSPEND
+                            }
+                            "callback" -> {
+                                invokeStyle = InvokeStyle.CALLBACK
+                            }
+                            "blocking" -> {
+                                invokeStyle = InvokeStyle.BLOCKING
+                            }
+                            else -> {
+                                throw RuntimeException("value for $arg option should be 'suspend', 'callback', or 'blocking'; instead got '$v'")
+                            }
+                        }
+                    }
+                    "-v" -> {
+                        verbosity = 1
+                    }
+                    "-vv" -> {
+                        verbosity = 2
+                    }
+                    "-vvv" -> {
+                        verbosity = 3
+                    }
+                    else -> {
+                        if (arg.startsWith("-")) {
+                            throw RuntimeException("unknown option: $arg")
+                        } else {
+                            throw RuntimeException("this command does not accept positional arguments")
+                        }
+                    }
+                }
+            }
+            return ClientArgs(invokeStyle, verbosity)
```

Comment:
This works for now but in the future I've had good success with https://args4j.kohsuke.org/ if we need more complexity here.

Reply:
Yeah, I've used a similar library, JCommander. I actually briefly looked into Clickt, which seems to be the most popular library for Kotlin command-line apps. But I had an issue with it (likely something quirky in the IDE, though I did update the project after adding a dep in gradle, but it still wouldn't work), so I decided to punt on pulling in a dep for this, since this logic is so simple 🤷